src/main/java/net/sf/jabref/model/database/BibDatabaseContext.java

```diff
@@ -176,6 +176,7 @@ public boolean isBiblatexMode() {
     * @return The default directory for this field type.
     */
     public List<String> getFileDirectories(String fieldName, FileDirectoryPreferences preferences) {
+        // TODO: all of these folder will be also searched for local PDFs. If this includes large folders like user.home this might take forever.
```

Comment:
Yes but it uses this functionality. The reason is, I added the user.dir as a 5th option to avoid the error message when a DB is not saved yet and the preferences are resetted (see #1446) but this lead to a "endless" looping of the find unlinked files because of the amount of files that typically reside inside the user.dir. This is more or less a hint what changes here can currently cause. I understand that this is not totally intuitive to anyone right now.